pkg/cmd/roachtest/roachtestutil/mixedversion/planner.go

```diff
@@ -1394,6 +1416,19 @@ func (ss stepSelector) CutBefore(predicate func(*singleStep) bool) (stepSelector
 	return append(before, cut...), after
 }
 
+// MarkNodesUnavailable marks all steps in the given step selector
+// as being in the context of a failure injection.
+func (ss stepSelector) MarkNodesUnavailable(systemUnavailable bool, tenantUnavailable bool) {
+	for _, s := range ss {
+		if systemUnavailable {
+			s.context.System.hasUnavailableNodes = true
+		}
+		if tenantUnavailable {
+			s.context.Tenant.hasUnavailableNodes = true
```

Comment:
I think you want to check if the tenant is nil here in case its a system only deployment? For example a vm reset failure (which isn't implemented yet), would take down both the system and the tenant so we pass true for both, but we only actually want to mark it if there is a tenant.